cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -135,6 +146,10 @@ std::optional<compute::Expression> ColumnChunkStatisticsAsExpression(
     min = maybe_min.MoveValueUnsafe();
     max = maybe_max.MoveValueUnsafe();
 
+    if (isNan(*min) && isNan(*max)) {
```

Comment:
Please add a comment explaining the reasons why NaN needs to be handled explicitly.

Reply:
Added a comment mentioning the NaN handling.